### PR TITLE
[Communication] - Remove dependency on Netty

### PR DESCRIPTION
**Pull request by paolamvhz**

### CODE REVIEW

**Review by ankitarorabit — CHANGES_REQUESTED** [review]
Sorry, should have been more specific with the Task name, the dependency needs to be removed from all packages (not just identity). If you remove the actual dependency from the Pom.xml of common, you will see that all packages will start failing and will need similar changes like you did for identity.

File: sdk/communication/azure-communication-chat/README.md [hunk 1]
```diff
@@ -69,24 +69,18 @@ The following sections provide several code snippets covering some of the most c
...
-<!-- embedme ./src/samples/java/com/azure/communication/chat/ReadmeSamples.java#L41-L57 -->
+<!-- embedme ./src/samples/java/com/azure/communication/chat/ReadmeSamples.java#L38-L48 -->
 ```Java
 String endpoint = "https://<RESOURCE_NAME>.communcationservices.azure.com";
 
-// Create an HttpClient builder of your choice and customize it
-// Use com.azure.core.http.netty.NettyAsyncHttpClientBuilder if that suits your needs
-NettyAsyncHttpClientBuilder httpClientBuilder = new NettyAsyncHttpClientBuilder();
-HttpClient httpClient = httpClientBuilder.build();
-
```
Comment (JianpingChen) [inline sdk/communication/azure-communication-chat/README.md:80]: The sample would not work if we remove this since the builder will throw if httpClient is not passed in. It'll be better we say this is just an example and Any implementation of HttpClient will work but it cannot be null

File: sdk/communication/azure-communication-chat/src/samples/java/com/azure/communication/chat/ReadmeSamples.java [hunk 2]
```diff
@@ -40,20 +37,14 @@
...
         String token = "SECRET";
         CommunicationTokenCredential credential = new CommunicationTokenCredential(token);
 
         // Initialize the chat client
         final ChatClientBuilder builder = new ChatClientBuilder();
         builder.endpoint(endpoint)
-            .credential(credential)
-            .httpClient(httpClient);
+            .credential(credential);
         ChatClient chatClient = builder.buildClient();
```
Comment (JianpingChen) [inline sdk/communication/azure-communication-chat/src/samples/java/com/azure/communication/chat/ReadmeSamples.java:48]: This line will throw

**Review by JianpingChen — CHANGES_REQUESTED** [review]

File: sdk/communication/azure-communication-chat/README.md (hunk 1 above)
Comment (ankitarorabit) [reply sdk/communication/azure-communication-chat/README.md:80]: This seems to have been taken care now, the httpClient requirement has been removed from the builder

File: sdk/communication/azure-communication-chat/src/samples/java/com/azure/communication/chat/ReadmeSamples.java (hunk 2 above)
Comment (ankitarorabit) [reply sdk/communication/azure-communication-chat/src/samples/java/com/azure/communication/chat/ReadmeSamples.java:48]: This is now fixed so it will not throw anymore

**Review by ankitarorabit — APPROVED** [review]

**Review by DominikMe — APPROVED** [review]

File: sdk/communication/azure-communication-common/pom.xml [hunk 3]
```diff
@@ -48,12 +48,6 @@
       <artifactId>azure-core</artifactId>
       <version>1.14.1</version> <!-- {x-version-update;com.azure:azure-core;dependency} -->
     </dependency>
-    <dependency>
-      <groupId>com.azure</groupId>
-      <artifactId>azure-core-http-netty</artifactId>
-      <version>1.9.0</version> <!-- {x-version-update;com.azure:azure-core-http-netty;dependency} -->
-      <scope>compile</scope>
-    </dependency>
```
Comment (srnagar) [inline sdk/communication/azure-communication-common/pom.xml:56]: Why is this removed? We should have netty as the default http client. 

By removing this from common, each of the other communication client libraries like chat, phone number etc will have to include this dependency.
